applications/openshift/api-server/api_server_openshift_https_serving_cert/rule.yml

```diff
@@ -28,27 +27,22 @@ references:
     srg: SRG-APP-000516-CTR-001325,SRG-APP-000516-CTR-001330,SRG-APP-000516-CTR-001335
 
 ocil_clause: |-
-    The openshift-apiserver serving-cert is not set to type
-    <tt>kubernetes.io/tls</tt> and that returned Data does not include <tt>tls.crt</tt>
-    and <tt>tls.key</tt>
+    The openshift-apiserver TLS security profile is set to old.
 
 ocil: |-
     Run the following command:
-    <pre>$ oc -n openshift-apiserver describe secret serving-cert</pre>
-    Verify that the <tt>serving-cert</tt> for the openshift-apiserver is type
-    <tt>kubernetes.io/tls</tt> and that returned Data includes <tt>tls.crt</tt>
-    and <tt>tls.key</tt>.
-
-# (jhrozek): Disabled because the compliance operator does not have the permissions
-#            to read secrets from openshift-apiserver
-#            - checking for the type should be enough as the type enforces the required
-#              keys
-# template:
-#     name: yamlfile_value
-#     vars:
-#         ocp_data: "true"
-#         filepath: '/api/v1/namespaces/openshift-apiserver/secrets/serving-cert'
-#         yamlpath: '.type'
-#         values:
-#             - value: 'kubernetes.io/tls'
-#               type: "string"
+    <pre>$ oc get APIServer cluster -o yaml</pre>
+    Verify that the <tt>tlsSecurityProfile</tt> is not type <tt>Old</tt>.
+
+warnings:
+- general: |-
+    {{{ openshift_cluster_setting("/apis/config.openshift.io/v1/apiservers/cluster") | indent(4) }}}
+
+template:
+  name: yamlfile_value
+  entity_check: none satisfy
+  vars:
+    ocp_data: "true"
+    filepath: /apis/config.openshift.io/v1/apiservers/cluster
+    yamlpath: "spec.tlsSecurityProfile.type"
```

Comment:
I'm not sure about the `entity_check`, doesn't it correspond to the object as opposed to the path? By default the object doesn't contain `spec.tlsSecurityProfile.type` and that's why the rule fails. Locally it helped to add `check_existence: "any_exist"`, but I'm not sure if that is the right thing.